src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = resize();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index!");
+        }
+        if (size == elementData.length) {
+            elementData = resize();
+        }
+        System.arraycopy(elementData, index, elementData, index + 1,
+                size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() >= elementData.length) {
+            elementData = resize();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (checkIndex(index)) {
+            return elementData[index];
+        }
+        throw new ArrayIndexOutOfBoundsException("Wrong parameter!");
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index!");
+        }
+        elementData[index] = value;
```

Comment:
how does this logic differ from that in `checkIndex`?
why can't we use it here?

Reply:
done